pkg/controllers/pushsecret/pushsecret_controller.go

```diff
@@ -72,9 +72,43 @@ type Reconciler struct {
 	ControllerClass string
 }
 
-func (r *Reconciler) SetupWithManager(mgr ctrl.Manager, opts controller.Options) error {
+func (r *Reconciler) SetupWithManager(ctx context.Context, mgr ctrl.Manager, opts controller.Options) error {
 	r.recorder = mgr.GetEventRecorderFor("pushsecret")
 
+	// Index PushSecrets by the stores they have pushed to (for finalizer management)
+	if err := mgr.GetFieldIndexer().IndexField(ctx, &esapi.PushSecret{}, "status.syncedPushSecrets", func(obj client.Object) []string {
+		ps := obj.(*esapi.PushSecret)
+		var storeNames []string
+
+		// Only index PushSecrets with DeletionPolicy=Delete for efficiency
+		if ps.Spec.DeletionPolicy != esapi.PushSecretDeletionPolicyDelete {
+			return storeNames
+		}
+
+		// Extract store names from syncedPushSecrets map
+		// Format is typically "Kind/Name" (e.g., "SecretStore/store1", "ClusterSecretStore/clusterstore1")
+		for storeKey := range ps.Status.SyncedPushSecrets {
+			// Extract just the store name from the key
+			if strings.Contains(storeKey, "/") {
```

Comment:
Efficiency and clarity: Consider using strings.Cut instead of Contains + SplitN.

```
var keys []string
	for storeKey := range ps.Status.SyncedPushSecrets {
		kind, name, found := strings.Cut(storeKey, "/")
		if found {
			// safer to keep full key "Kind/Name" instead of just name
			keys = append(keys, kind+"/"+name)
		}
	}
```